### PR TITLE
Opus fix

### PR DESCRIPTION
**Pull request by drnikolaev**
Fixes client crash for unsupported OPUS sample rates

### CODE REVIEW

**Comment by virajkarandikar** [comment]
Can you also include fix for print at https://github.com/nvidia-riva/cpp-clients/blob/main/riva/utils/opus/opus_decoder.cc#L141 ?

**Review by shikshapatel — COMMENTED** [review]
lgtm

**Comment by drnikolaev** [comment]
> Can you also include fix for print at https://github.com/nvidia-riva/cpp-clients/blob/main/riva/utils/opus/opus_decoder.cc#L141 ?

What exactly should be fixed there?

File: riva/clients/tts/riva_tts_client.cc [hunk 1]
```diff
@@ -198,7 +200,8 @@ main(int argc, char** argv)
     if (FLAGS_audio_encoding.empty() || FLAGS_audio_encoding == "pcm") {
       ::riva::utils::wav::Write(FLAGS_audio_file, FLAGS_rate, pcm_buffer.data(), pcm_buffer.size());
```
Comment (virajkarandikar) [inline riva/clients/tts/riva_tts_client.cc:201]: wav header rate also needs to be updated.

File: riva/clients/tts/riva_tts_client.cc [hunk 2]
```diff
@@ -198,7 +200,8 @@ main(int argc, char** argv)
...
     } else if (FLAGS_audio_encoding == "opus") {
-      riva::utils::opus::Decoder decoder(FLAGS_rate, 1);
+      int32_t rate = riva::utils::opus::Encoder::AdjustRateIfUnsupported(FLAGS_rate);
+      riva::utils::opus::Decoder decoder(rate, 1);
       auto pcm = decoder.DecodePcm(decoder.DeserializeOpus(opus_buffer));
       ::riva::utils::wav::Write(FLAGS_audio_file, FLAGS_rate, pcm.data(), pcm.size());
```
Comment (virajkarandikar) [inline riva/clients/tts/riva_tts_client.cc:206]: wav header rate also needs to be updated.

File: riva/clients/tts/riva_tts_client.cc (hunk 1 above)
Comment (virajkarandikar) [reply riva/clients/tts/riva_tts_client.cc:201]: if user runs without any rate param, FLAGS_rate is 44100. But rate is 24000.

**Comment by virajkarandikar** [comment]
Closing stale PR. Fixes have been merged.

**Pull request closed without merging**